### DISCUSSION TITLE
Yi Home 1080 IFUS - FTP UPLOAD DOESN'T WORK

### DISCUSSION THREAD

**Discussion by Giuserver** [roleoroleo/yi-hack-Allwinner-v2#336] [Q&A] [answered]
Hello everybody, i have 6 camera's (3 BUFS e 3 IFUS). The BFUS camera's upload files regularly on FTP (local) Server. The IFUS camera's upload just the first file on record directory. Any idea from this bug on IFUS cameras?  
I've compared ftppush.sh file from both type but they have same code! Now i mount y211ga 0.1.9

from the command ash /tmp/sd/yi-hack/script/ftppush.sh cron
i've this
 2022-01-16 [17-01-40] [INFO] checkFiles: ignore file [/tmp/sd/record/2022Y01M16D17H/E100M00S60.mp4] - already sent.
 2022-01-16 [17-01-40] [INFO] checkFiles: ignore file [/tmp/sd/record/2022Y01M16D17H/E101M00S60.mp4] - already sent.
2022-01-16 [17-01-40] [INFO] checkFiles: ignore file [/tmp/sd/record/2022Y01M16D17H/E102M00S60.mp4] - already sent.

i found an error in ftppush.sh 
this script doesn't write correctly the time in the 'last_file_sent' file
@roleoroleo can you help me? Thanks

**Comment by roleoroleo** [comment]
Please, explain me better the sentence "this script doesn't write correctly the time in the 'last_file_sent' file".

**Reply by Giuserver** [reply]
I've modified the file yi-hack/scripts/ftppush.sh in:

```

checkFiles ()
{
	#
	FTP_FILE_DELETE_AFTER_UPLOAD="$(get_config FTP_FILE_DELETE_AFTER_UPLOAD)"
	#
	logAdd "[INFO] checkFiles"
	#
	# Search for new files.
	if [ -f "/usr/bin/sort" ] || [ -f "/tmp/sd/yi-hack/usr/bin/sort" ]; then
		# Default: Optimized for busybox
		L_FILE_LIST="$(find "${FOLDER_TO_WATCH}" -mindepth ${FOLDER_MINDEPTH} -type f \( -name "${FILE_WATCH_PATTERN}" \) | sort -k 1 -n)"
	else
		# Alternative: Unsorted output
		L_FILE_LIST="$(find "${FOLDER_TO_WATCH}" -mindepth ${FOLDER_MINDEPTH} -type f \( -name "${FILE_WATCH_PATTERN}" \))"
	fi
	if [ -z "${L_FILE_LIST}" ]; then
		return 0
	fi
	#
	echo "${L_FILE_LIST}" | while read file; do
		if [ "${#file}" == "14" ]; then
			FILE_DATE=$(date +%Y-%m-%dT%H-%M)
		else
			FILE_DATE=$(date +%Y-%m-%dT%H-%M)
		fi
		FILE_YEAR=${FILE_DATE:0:4}
		FILE_REMPART=${FILE_DATE:5:2}${FILE_DATE:8:2}${FILE_DATE:11:2}${FILE_DATE:14:2}
		LAST_FILE_SENT=$(cat /tmp/last_file_sent)
		LAST_FILE_SENT_YEAR=${LAST_FILE_SENT:0:4}
		LAST_FILE_SENT_REMPART=${LAST_FILE_SENT:5:2}${LAST_FILE_SENT:8:2}${LAST_FILE_SENT:11:2}${LAST_FILE_SENT:14:2}
		if [ ${FILE_YEAR} -gt ${LAST_FILE_SENT_YEAR} ] || ( [ ${FILE_YEAR} -eq ${LAST_FILE_SENT_YEAR} ] && [ ${FILE_REMPART} -gt ${LAST_FILE_SENT_REMPART} ] ); then
			if ( ! uploadToFtp -- "${file}" ); then
				logAdd "[ERROR] checkFiles: uploadToFtp FAILED - [${file}]."
				continue
			fi
			logAdd "[INFO] checkFiles: uploadToFtp SUCCEEDED - [${file}]."
			LAST_FILE_SENT=${FILE_DATE}
			echo $LAST_FILE_SENT > ${LAST_FILE_SENT_FILE}
			sync
			if [ "${FTP_FILE_DELETE_AFTER_UPLOAD}" == "yes" ]; then
				rm -f "${file}"
			fi
		else
			logAdd "[INFO] checkFiles: ignore file [${file}] - already sent."
			logAdd "Gisueppe [${FILE_DATE}] "
		fi
		#
	done
	#
	# Delete empty sub directories
	if [ ! -z "${FOLDER_TO_WATCH}" ]; then
		for d in $(find "${FOLDER_TO_WATCH}/" -mindepth 1 -type d); do
			#find "${FOLDER_TO_WATCH}/" -mindepth 1 -type d -empty -delete
			[ -z "`find $d -type f`" ] && rmdir $d
		done
	fi
	#
	return 0
}

```

Now works fine

**Comment by roleoroleo** [comment]
I tested again the script on my cam and I can't reproduce the problem you noticed.
My cam saves the date correctly.
Are you using "FTP Delete (after upload)"?

**Reply by Giuserver** [reply]
i've tried both, but nothing.. same problem

**Comment by ostap7474** [comment, marked as answer]
I have the same problem on an IFUS camera.

**Reply by Giuserver** [reply]
if you wanna solve just open the file yi-hack/scripts/ftppush.sh  
find he row
`echo "${L_FILE_LIST}" | while read file; do `
and below, modified the row in if else with:
`FILE_DATE=$(date +%Y-%m-%dT%H-%M)`

then upload all files in camera and works

**Reply by ostap7474** [reply]
Now everything works. Thank you.
